api/test/models/attendance/attendence-data-mapper-impl.test.ts

```diff
@@ -0,0 +1,166 @@
+// GLOBAL REQUIREMENTS
+import { Substitute } from '@fluffy-spoon/substitute';
+import { expect } from 'chai';
+import 'mocha';
+import { of } from 'rxjs';
+import { IActiveHackathonDataMapper } from '../../../lib/models/hackathon/active-hackathon';
+import { ActiveHackathon } from '../../../lib/models/hackathon/active-hackathon/active-hackathon';
+import { Attendance,
+         IAttendanceDataMapper,
+         AttendanceDataMapperImpl} from '../../../lib/models/attendance'
+import { RBAC } from '../../../lib/services/auth/RBAC/rbac';
+import { IAcl } from '../../../lib/services/auth/RBAC/rbac-types';
+import { MysqlUow } from '../../../lib/services/database/svc/mysql-uow.service';
+import { Logger } from '../../../lib/services/logging/logging';
+
+function mockedQuery<T>(query, params) {
+  // @ts-ignore
+  return Promise.resolve({ query, params });
+}
+
+let attendanceDataMapper: IAttendanceDataMapper;
+let activeHackathonDataMapper;
+let mysqlUow;
+const acl: IAcl = new RBAC();
+
+describe('TEST: Attendance data mapper', () => {
+  beforeEach(() => {
+    // Configure Mock MysqlUow
+    mysqlUow = Substitute.for<MysqlUow>();
+    // Configure Active Hackathon Data Mapper
+    activeHackathonDataMapper = Substitute.for<IActiveHackathonDataMapper>();
+    activeHackathonDataMapper.activeHackathon.returns(of(new ActiveHackathon({
+      basePin: 0,
+      endTime: null,
+      name: 'test hackathon',
+      uid: 'test uid',
+    })));
+    activeHackathonDataMapper.tableName.returns('HACKATHON');
+    // Configure Attendance Data Mapper
+    attendanceDataMapper = new AttendanceDataMapperImpl(
+      acl,
+      mysqlUow,
+      activeHackathonDataMapper,
```

Comment:
Does attendaceDataMapper need activeHackathon dependence?

Reply:
Yeah getAll and getCount for the AttendanceDataMapperImpl both use it.